integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java

```diff
@@ -74,16 +74,16 @@
             featureBeans.add((Bean< ? extends Feature >)event.getBean());
         } else if (CdiBusBean.CXF.equals(event.getBean().getName())
                 && Bus.class.isAssignableFrom(event.getBean().getBeanClass())) {
-            busBean = event.getBean();
+            hasBus = true;
         }
     }
 
     public void load(@Observes final AfterDeploymentValidation event, final BeanManager beanManager) {
+        // no need of creational context, it only works for app scoped instances anyway
+        final Bean<?> busBean = beanManager.resolve(beanManager.getBeans(CdiBusBean.CXF));
         bus = (Bus)beanManager.getReference(
```

Comment:
@johnament this is not the iterator since there is no specialization and a single item in the list, issue is really linked to the *instance* (as object instance) of the Bean<?>. If user does event.addBean(new MyJAXRSBean()) then you will have it as well. Surely less impacting cause you will get one instance in CXF and one in the app if the resource is injected...which never happens in practise.